securedrop-admin

```diff
@@ -0,0 +1,120 @@
+
+def activate_venv():
+    """Use to activate the local virtualenv"""
+    try:
+        execfile(VENV_ACTIVATION, dict(__file__=VENV_ACTIVATION))
+    except FileNotFoundError:
+        error("Pre-requisites not in place, re-run command with 'setup'",
+                level='error')
+
+def sdconfig(args):
+    """Configure SD site settings"""
+    activate_venv()
+    subprocess.call(['./configure'], cwd=ANSIBLE_PATH, env=os.environ.copy())
+
+def envsetup(args):
+    """Install SD pre-reqs"""
+    # virtualenv doesnt exist? Install dependencies and create
+    if not os.path.exists(VENV_ACTIVATION):
+        subprocess.call(['sudo', 'su', '-c', 
+                        "apt-get update && \
+                        apt-get install -y \
+                        python-virtualenv \
+                        python-pip \
+                        ccontrol \
+                        virtualenv \
+                        libffi-dev \
+                        libssl-dev \
+                        libpython2.7-dev"])
+
+        # Technically you can create a virtualenv from within python
+        # but pip can only be run over tor on tails, and debugging that
+        # along with instaling a third-party dependency is not worth
+        # the effort here.
+        subprocess.call(['torify', 'virtualenv', VENV_DIR])
+
+    pip_install_cmd = ['torify', os.path.join(VENV_DIR, 'bin', 'pip'), 'install']
+    requirements_cmd = ['-r', os.path.join(ANSIBLE_PATH, 'requirements.txt'),
+                        '--require-hashes']
+
+    if args.update:
+        subprocess.call(pip_install_cmd+['-U']+requirements_cmd)
+    else:
+        subprocess.call(pip_install_cmd+requirements_cmd)
+
+def install_securedrop(args):
+    """Install/Update SecureDrop"""
+    activate_venv()
+    # Yaml library cannot be imported until virtualenv is activated
+    # (hence the yaml library is not imported up-top)
+    import yaml
+
+    try:
+        site_config_vars = yaml.safe_load(open(SITE_CONFIG, 'ro').read())
+
+        ansible_user = args.user if args.user else site_config_vars['ssh_users']
+
+        sudo_prompt = getpass.getpass(prompt="Sudo Pass [Blank if not needed]:")
+        subprocess.call(['./securedrop-prod.yml',
+                '-u '+ansible_user,
+                "-e ansible_become_pass={}".format(sudo_prompt if sudo_prompt else ""),
+                ],
+                cwd=ANSIBLE_PATH)
+    except IOError:
+        error("Config file missing, re-run with sdconfig", level='error')
+
+def run_tails_config(args):
+    """Configure Tails environment post SD install"""
+    subprocess.call(['sudo','./tails_files/install.sh'])
+
+if __name__ == "__main__":
+    # Processing argument parsing logic -- yuck
+    parser = argparse.ArgumentParser(description = __doc__)
+    subparsers = parser.add_subparsers()
+
+    parse_setup = subparsers.add_parser('setup', help=envsetup.__doc__)
+    parse_setup.add_argument('--update', action='store_true', default=False,
```

Comment:
This should be `--upgrade` to match the corresponding name of the `pip` flag. However, I think that instead of making this an optional switch, we should _always_ include the `-U` flag. Since a `.txt` file generated by `pip-compile` is a self-contained dependency chain in which all package versions are fixed, `pip install -U -r requirements.txt` will always be an idempotent operation until the `.txt` file is actually changed (as opposed to the case where only top-level dependencies are listed/ have their versions fixed in a `.txt` file, wherein it is possible lower-level dependencies get upgraded over time).

This should make the virtualenv update process slightly easier for the admins too.